src/plugins/discover/public/application/components/top_nav/get_top_nav_links.tsx

```diff
@@ -194,7 +192,7 @@ export const getTopNavLinks = (
       share?.toggleShareContextMenu({
         anchorElement,
         allowEmbed: false,
-        allowShortUrl: capabilities.discover.createShortUrl as boolean,
+        allowShortUrl: capabilities.discover?.createShortUrl as boolean,
```

Comment:
How do we determine whether short urls are allowed?

Reply:
Sorry @zashary but what do you mean by that? I didnt understand the question. 

The change that you highlighted here was added for type safety. While discover should always have a boolean value for `createShortUrl`, The capability service as it stands today is not strictly typed for that and this property can be left undefined if initialized differently. Instead of refactoring the whole capability service, i just corrected the type for the capability service so that every use of the service can handle for an unexpected undefined if that happens.